### PR TITLE
feat: update cluster API library to the latest version

### PR DESCRIPTION
**Pull request by Unix4ever**
This is necessary to get rid of dependency hell and update to v1alpha4.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,15 +3,15 @@ module github.com/talos-systems/capi-utils
 go 1.16
 
 require (
-	github.com/spf13/cobra v1.1.3
-	github.com/spf13/viper v1.7.0
+	github.com/spf13/cobra v1.2.1
+	github.com/spf13/viper v1.8.1
 	github.com/talos-systems/go-debug v0.2.1
 	github.com/talos-systems/go-retry v0.3.1
 	github.com/talos-systems/talos/pkg/machinery v0.11.5
```
Comment (smira) [inline go.mod:10]: we could also bump Talos machinery? or it doesn't matter now?

**Review by smira — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (Unix4ever) [reply go.mod:10]: surely doesn't hurt to bump it :+1:

**Review by smira — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**